R/check.R

```diff
@@ -402,6 +402,39 @@ checkPlots <- function(plots) {
   return(NULL)
 }
 
+checkMapping <- function(mapping) {
+  checkList(mapping)
+
+  # stop if mapping object has less than 2 elements
+  if (length(mapping) > 0) stopifnot(length(mapping) > 1)
+  else return(NULL)
+
+  # check if list elements have the same size. If not, fill difference with NA.
+  listMaxLength <- max(sapply(mapping, length))
+  mapping <- lapply(lapply(mapping, unlist), "length<-", listMaxLength)
+
+  mappingdf <- as.data.frame(mapping)
```

Comment:
I figured out the error that only occurs with R 3.4.4. In R3, strings are always automatically coerced to factors by default (`stringsAsFactors=TRUE`). This default behavior was changed (for the better) in R4. Please update this line to explicitly keep the strings as charactor vectors when converting to a data frame.

```R
 mappingdf <- as.data.frame(mapping, stringsAsFactors = FALSE)
```